src/MongoDB/WriteResult.c

```diff
@@ -136,8 +142,9 @@ PHP_METHOD(WriteResult, getUpsertedCount)
 		return;
 	}
 
-
-	RETURN_LONG(intern->write_result.nUpserted);
+	if (bson_iter_init_find(&iter, intern->reply, "nUpserted") && BSON_ITER_HOLDS_INT32(&iter)) {
+		RETURN_LONG(bson_iter_int32(&iter));
+	}
```

Comment:
The code with this iter_find, and RETURN_LONG is repeat 4/5 times. Wouldn't it make more sense to make a function (or macro) for this?